### PR TITLE
Fix a number of ShellCheck warnings

### PR DESCRIPTION
**Pull request by sicherha**
I have put the script through ShellCheck and fixed the most conspicuous warnings - in particular those related to quoting, as unintended globbing and word splitting can lead to unforeseen consequences.

### CODE REVIEW

File: nerdps1 [hunk 1]
```diff
@@ -642,9 +642,9 @@ function pssh
     [ "$1" ] && [ -r "$1" ] && customenv=$1 && shift
     case "$sshmode" in
         1|raw) ssh -t "$remote" "$@" "$(ps1_sshcmd "$loginopts" "$customenv")";;
-        2|b64) ssh -t "$remote" "$@" ". <(echo $(ps1_sshcmd "$loginopts" "$customenv"| gzip -c| base64 $ps1_b64opt) |base64 -d|gzip -dc)";;
+        2|b64) ssh -t "$remote" "$@" ". <(echo $(ps1_sshcmd "$loginopts" "$customenv"| gzip -c| base64 "$ps1_b64opt") |base64 -d|gzip -dc)";;
```
Comment (joknarf) [inline nerdps1]:
```suggestion
        2|b64) ssh -t "$remote" "$@" ". <(echo $(ps1_sshcmd "$loginopts" "$customenv"| gzip -c| base64 $ps1_b64opt) |base64 -d|gzip -dc)";;
```

File: nerdps1 [hunk 2]
```diff
@@ -655,21 +655,21 @@ psshu() { pssh --url "$@"; }
 function ps1_pssh3 {
     typeset remote="$1" loginopts="$2" customenv="$3" ssh_config=$TMPDIR/.nerdps1.$USER.ssh argf=false
     shift 3
-    (umask 077;echo "RemoteCommand . <(echo $(ps1_sshcmd "$loginopts" "$customenv"| gzip -c| base64 $ps1_b64opt) |base64 -d|gzip -dc)" >$ssh_config) || return 1
+    (umask 077;echo "RemoteCommand . <(echo $(ps1_sshcmd "$loginopts" "$customenv"| gzip -c| base64 "$ps1_b64opt") |base64 -d|gzip -dc)" >"$ssh_config") || return 1
```
Comment (joknarf) [inline nerdps1]:
```suggestion
    (umask 077;echo "RemoteCommand . <(echo $(ps1_sshcmd "$loginopts" "$customenv"| gzip -c| base64 $ps1_b64opt) |base64 -d|gzip -dc)" >"$ssh_config") || return 1
```

**Review by joknarf — CHANGES_REQUESTED** [review]
Hello, 
Thanks for your work,
Just a regression on `$ps1_b64opt` that must not be quoted, as some version of base64 must have no option (and not empty option), typically macOS base64.

File: nerdps1 (hunk 1 above)
Comment (sicherha) [reply nerdps1]: Done.

File: nerdps1 (hunk 2 above)
Comment (sicherha) [reply nerdps1]: Done.

**Comment by sicherha** [comment]
> Hello, Thanks for your work, Just a regression on `$ps1_b64opt` that must not be quoted, as some version of base64 must have no option (and not empty option), typically macOS base64.

Good catch. I wonder if it would make sense to turn `ps1_b64opt` and `ps1_info_vars` from space-separated strings into arrays. While I'm not overly familiar with ksh, it seems that all three of Bash, Zsh and ksh do support the `"${myarray[@]}"` notation for expanding an array into an argument list while maintaining proper quoting per element.

**Comment by joknarf** [comment]
> > Hello, Thanks for your work, Just a regression on `$ps1_b64opt` that must not be quoted, as some version of base64 must have no option (and not empty option), typically macOS base64.
> 
> Good catch. I wonder if it would make sense to turn `ps1_b64opt` and `ps1_info_vars` from space-separated strings into arrays. While I'm not overly familiar with ksh, it seems that all three of Bash, Zsh and ksh do support the `"${myarray[@]}"` notation for expanding an array into an argument list while maintaining proper quoting per element.

very good idea indeed, but losing busybox ash compatibility as no array support in ash

**Review by joknarf — APPROVED** [review]

**Pull request merged**